README.md

```diff
@@ -13,12 +13,17 @@
 
 `avail-light` is a data availability light client that can do the following:
 
-* Listen for newly produced blocks.
+* Listen for newly mined blocks
 * Gain confidence for `N` *cells*, where *cell* is defined as a `{row, col}` pair.
-  > As soon as a new block is available, the light client attempts to gain confidence by asking for a proof from a full 
-  client via JSON-RPC.
-* Batch processces blocks in reverse order (i.e. prioritizes the latest blocks) for lower numbered blocks where no confidence 
-  is yet gained.
+  > As soon as a new block is available, the light client attempts to gain confidence by asking for a proof from a full client via JSON-RPC.
+* Batch processces blocks in reverse order (i.e. prioritizes the latest blocks) for lower numbered blocks 	where no confidence is yet gained.
+
+### Modes of Operation
+
+1. **Light client**: This is the basic mode of operation and is always active in whichever mode it works. If the APP_ID is not given then this mode will be commence. In this mode, the client on each header it receieves will do the random sampling using RPC calls. It gets random cells in return, which verifies and calculate the confidence.
+2. **App-Specific-Mode**: In this mode if an App_ID(greater than 0) is given in the config file then the client would find out the cols related the App_ID given using the app_data_lookup in the header. Then it fetches the 50% of cells from the cols, verifies them and uses them to decode the app_extrinsics_data. 
+3. **Fat-Client-Mode**: In this mode, the client gets the entire extended matrix using IPFS if it is available or uses RPC calls to fetch. It verifies the entire cells and computes the CID mapping for the IPFS Pinning. It then decodes the entire matrix and reconstruct the entire app_specific_data related to all App_IDs.
```

Comment:
`entire cells` -> `all the cells` or `entire matrix`? 
Decodes matrix -> reconstructs original matrix or decodes extended matrix?

Reply:
I think its original matrix

Reply:
@aterentic-ethernal